frontend/lib/src/components/widgets/Slider/Slider.tsx

```diff
@@ -282,159 +149,294 @@ class Slider extends React.PureComponent<Props, State> {
-        "onMouseLeave",
-        "draggable",
-      ])
-      const ariaValueText: Record<string, string> = {}
-
-      if (this.props.element.options.length > 0 || this.isDateTimeType()) {
-        ariaValueText["aria-valuetext"] = formattedValue
+    thumbRefs.map((ref, i) => {
+      if (!ref.current) {
+        return
       }
+      ref.current.setAttribute("aria-valuetext", formattedValueArr[i])
+    })
+  })
 
-      return (
-        <StyledThumb
-          {...passThrough}
-          disabled={props.$disabled === true}
-          ref={this.thumbRef[thumbIndex]}
-          aria-valuetext={formattedValue}
-          aria-label={this.props.element.label}
-        >
-          <StyledThumbValue
-            data-testid="stSliderThumbValue"
-            disabled={props.$disabled === true}
-            ref={this.thumbValueRef[thumbIndex]}
-          >
-            {formattedValue}
-          </StyledThumbValue>
-        </StyledThumb>
-      )
+  // When resetting a form, `value` will change so we need to change `uiValue`
+  // to match.
+  useEffect(() => {
+    if (value !== uiValue) {
+      setUiValue(value)
     }
+    // Don't include `uiValue` in the deps below or the slider will become
+    // jittery.
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [value])
+
+  const debouncedSetValueWithSource = useCallback(
+    debounce(DEBOUNCE_TIME_MS, (value: number[]): void => {
+      setValueWithSource({ value, fromUi: true })
+    }),
+    [setValueWithSource]
   )
 
-  private renderTickBar = (): JSX.Element => {
-    const { disabled, element } = this.props
-    const { max, min } = element
+  const handleChange = useCallback(
+    ({ value }: { value: number[] }): void => {
+      setUiValue(value)
+      debouncedSetValueWithSource(value)
+    },
+    [setValueWithSource]
+  )
 
+  const renderTickBar = useCallback((): ReactElement => {
     return (
       <StyledTickBar data-testid="stSliderTickBar">
         <StyledTickBarItem
           disabled={disabled}
           data-testid="stSliderTickBarMin"
         >
-          {this.formatValue(min)}
+          {formattedMinValue}
         </StyledTickBarItem>
         <StyledTickBarItem
           disabled={disabled}
           data-testid="stSliderTickBarMax"
         >
-          {this.formatValue(max)}
+          {formattedMaxValue}
         </StyledTickBarItem>
       </StyledTickBar>
     )
-  }
-
-  public render(): React.ReactNode {
-    const { disabled, element, theme, width, widgetMgr } = this.props
-    const { colors, fonts, fontSizes, spacing } = theme
-    const style = { width }
-
-    // Manage our form-clear event handler.
-    this.formClearHelper.manageFormClearListener(
-      widgetMgr,
-      element.formId,
-      this.onFormCleared
-    )
+    // Only run this on first render. No real reason, just nicer.
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [])
+
+  const renderThumb = useCallback(
+    /* eslint-disable react/display-name */
```

Comment:
nit: we can get rid of disabling eslint here by doing this:

```typescript
    React.forwardRef<HTMLDivElement, StyleProps>(function renderThumb(
      props: StyleProps,
      ref
    ): ReactElement {
      const { $thumbIndex } = props
// keep rest like it is
```

I think the advantage is that upon errors we get a better trace with a defined name and we don't disable eslint 😁